app/src/tunnel/kotlin/tunnel/Persistence.kt

```diff
@@ -65,7 +75,8 @@ class FiltersPersistence {
     }()
 
     private fun loadLegacy35(ktx: AndroidKontext) = {
-        Result.of { core.Persistence.paper().read("filters2", FiltersCache()) }
```

Comment:
But anyone who loads this new app will be updated to new (4.0) version. For this code to take effect that would mess with old 3.5 version, it would mean they would have to be using version 4.0. Seems like it wouldn't matter. If anything, all the outdated code should be deleted because only way it comes into effect is if user updates app and the update wouldn't use it. Correct?

Reply:
The `loadLegacyXX()` are used to import old settings. If somebody has version 3.x installed and updates the app he expects his settings to be preserved. Every newer version is able to load settings from older versions and convert them into the newer form.